### ISSUE TITLE
[Bug]: Steering will sometimes make a sharp input

### ISSUE THREAD

**Issue by ChibiDreamweaver** [emoose/OutRun2006Tweaks#141]
### Tweaks version

outrun2006tweaks-9e79594a9fda217c32b1eb22023bd9ac3f66d61d

### Describe your issue here (please attach the OutRun2006Tweaks.log file with your report, you can drag+drop it into the text area)

Utilizing the game's default 0.2% steering deadzone, I have found that on occasion, the car will steer too sharply instead of reading the input correctly, sending me into a wall if I do not catch it in time. this had not happened in older versions (even in older official releases). This is not a controller issue and I know this because I have changed from a PowerA Xbox One Controller (PowerA is notorious for having bad 3rd party controllers and I had one with really bad stick drift leading to having my inputs not being read AT ALL for a few seconds) to a GameSir Xbox Series Controller (Which has no stick drift issues right out of the box yesterday).

This is ruining my OutRun 2SP mode time attack runs and this needs to be looked into before another official release comes out.

**Comment by ChibiDreamweaver** [comment]
UPDATE: Upon adjusting the deadzone to 0.1% in this specific version of Tweaks, the sudden sharp turns have stopped. something is wrong with the 0.2% percent deadzone. hopefully, this bug will be squashed before the next official release.

**Comment by emoose** [comment]
So this is something that only started in recent tweaks releases?

Not really sure why 0.2 deadzone would affect it, if it's set to 0.2 it should deactivate the deadzone patches since vanilla game uses 0.2 by default.

**Comment by ChibiDreamweaver** [comment]
> So this is something that only started in recent tweaks releases?
> 
> Not really sure why 0.2 deadzone would affect it, if it's set to 0.2 it should deactivate the deadzone patches since vanilla game uses 0.2 by default:
> 
> https://github.com/emoose/OutRun2006Tweaks/blob/df7fac4f10d51b18854f8da4db1f6f6e08faaac3/src/hooks_input.cpp#L333-L336
> 
> Could you check if https://github.com/emoose/OutRun2006Tweaks/releases/tag/v0.5.1 has the issue?

yes, apparently so. I'll get 5.1 in my game rn and see if it has the same issue

**Comment by ChibiDreamweaver** [comment]
Just tested version 0.5.1. Fortunately, this patch does not have the issue I have mentioned in this post. This is just a guess but it could be any version of Tweaks where the ability to change the steering deadzone is where the issue came from.

**Comment by emoose** [comment]
>This is just a guess but it could be any version of Tweaks where the ability to change the steering deadzone is where the issue came from.

Seems likely, but not really sure why since it should be getting deactivated at 0.2 like mentioned :/

Could you try with these 3 builds and see if there's any difference?

https://github.com/emoose/OutRun2006Tweaks/actions/runs/10511134416
https://github.com/emoose/OutRun2006Tweaks/actions/runs/10543165711
https://github.com/emoose/OutRun2006Tweaks/actions/runs/10551379579

The first one there is just prior to deadzone being added, middle one includes the deadzone stuff, and last one is just after deadzone was added.

May want to backup your INI files and use the INI included with them, since there were some INI changes in those.

**Comment by ChibiDreamweaver** [comment]
This may be the case of the bad controller since I just recently tried the 0.2 deadzone and it looks like it is a me issue lol
but I'll try the tweaks you just linked when I can

**Comment by k0pp0** [comment]
Would be interested in what happens here.

I have also experienced it, but put it down to CPU load affecting things (n100/win11).

Will try some longer tests in batocera/wine when I have time as the game runs better/looks better under wine for some reason.

**Comment by ChibiDreamweaver** [comment]
> Would be interested in what happens here.
> 
> I have also experienced it, but put it down to CPU load affecting things (n100/win11).
> 
> Will try some longer tests in batocera/wine when I have time as the game runs better/looks better under wine for some reason.

What other issues have you encountered? I sometimes encounter the occasional stutter but it's not the kind of stutter that will cause the game to hitch for a bit too long. I put that issue down as I probably need to expand my ram for my desktop so the game can run as smooth as it can

**Comment by ChibiDreamweaver** [comment]
> Would be interested in what happens here.
> 
> I have also experienced it, but put it down to CPU load affecting things (n100/win11).
> 
> Will try some longer tests in batocera/wine when I have time as the game runs better/looks better under wine for some reason.

also please keep us posted in case you find something that is the cause of our issue. I just want to know before I start trying the tweaks files Emoose posted in this thread

**Comment by ChibiDreamweaver** [comment]
To give an update, even with the 0.1% deadzone setting, I ran into the same issue in this thread. This is probably being caused by the CPU load that k0pp0 had mentioned earlier. The CPU load might be the culprit but who knows

**Comment by emoose** [comment]
Could try disabling SingleCoreAffinity, having that enabled does increase CPU load quite a bit afaik, might help remove the stutters but if you have a very modern CPU (I think ones with e-cores) it could cause freezes.

**Comment by k0pp0** [comment]
To be honest CPU load is so low on mine, even on a n100. GPU isn't stressed either

**Comment by ChibiDreamweaver** [comment]
> Could try disabling SingleCoreAffinity, having that enabled does increase CPU load quite a bit afaik, might help remove the stutters but if you have a very modern CPU (I think ones with e-cores) disabling it might cause freezes.

Will do
